presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraMetadata.java

```diff
@@ -185,7 +185,7 @@ private ConnectorTableMetadata getTableMetadata(SchemaTableName tableName)
 
     private List<SchemaTableName> listTables(ConnectorSession session, SchemaTablePrefix prefix)
     {
-        if (prefix.getSchemaName() == null) {
+        if (prefix.getTableName() == null) {
             return listTables(session, prefix.getSchemaName());
         }
         return ImmutableList.of(new SchemaTableName(prefix.getSchemaName(), prefix.getTableName()));
```

Comment:
What if `schemaName` is `null` here?

Reply:
At first, I thought when `schemaName` is `null`, `tableName` is also always `null` and ` if (prefix.getTableName() == null)` can handle such case.

SchemaTablePrefix.java
```java
    public SchemaTablePrefix()
    {
        this.schemaName = null;
        this.tableName = null;
    }

    public SchemaTablePrefix(String schemaName)
    {
        this.schemaName = checkNotEmpty(schemaName, "schemaName");
        this.tableName = null;
    }

    public SchemaTablePrefix(String schemaName, String tableName)
    {
        this.schemaName = checkNotEmpty(schemaName, "schemaName");
        this.tableName = checkNotEmpty(tableName, "tableName");
    }
```